conbench/api/runs.py

```diff
@@ -34,6 +37,30 @@ def get(self, run_id):
         run = self._get(run_id)
         return self.serializer.one.dump(run)
 
+    @flask_login.login_required
+    def delete(self, run_id):
+        """
+        ---
+        description: Delete a run.
+        responses:
+            "204": "204"
+            "401": "401"
+            "404": "404"
+        parameters:
+          - name: run_id
+            in: path
+            schema:
+                type: string
+        tags:
+          - Runs
+        """
+        summaries = Summary.all(run_id=run_id)
```

Comment:
As mentioned in slack, we should cascade on delete rather than first deleting each benchmark result in a run.

https://docs.sqlalchemy.org/en/14/orm/cascades.html#delete

This can be done in a follow-up pull request though.